flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/sink/dynamic/TableMetadataCache.java

```diff
@@ -47,16 +48,24 @@ class TableMetadataCache {
   private static final int MAX_SCHEMA_COMPARISON_RESULTS_TO_CACHE = 10;
   private static final Tuple2<Boolean, Exception> EXISTS = Tuple2.of(true, null);
   private static final Tuple2<Boolean, Exception> NOT_EXISTS = Tuple2.of(false, null);
-  static final Tuple2<Schema, CompareSchemasVisitor.Result> NOT_FOUND =
-      Tuple2.of(null, CompareSchemasVisitor.Result.SCHEMA_UPDATE_NEEDED);
+  static final SchemaCompareInfo NOT_FOUND =
+      new SchemaCompareInfo(
+          null, CompareSchemasVisitor.Result.SCHEMA_UPDATE_NEEDED, DataConverter.identity());
 
   private final Catalog catalog;
   private final long refreshMs;
+  private final int inputSchemasPerTableCacheMaximumSize;
   private final Cache<TableIdentifier, CacheItem> cache;
 
   TableMetadataCache(Catalog catalog, int maximumSize, long refreshMs) {
+    this(catalog, maximumSize, refreshMs, MAX_SCHEMA_COMPARISON_RESULTS_TO_CACHE);
+  }
```

Comment:
Can we remove this constructor or annotate with `@VisibleForTesting`? I think its best to remove it to avoid confusion.